source/common/upstream/upstream_impl.h

```diff
@@ -673,8 +672,9 @@ class ClusterImplBase : public Cluster, protected Logger::Loggable<Logger::Id::u
    */
   void onInitDone();
 
+  Init::ManagerImpl init_manager_;
```

Comment:
From a quick skim it's a little confusing when we have a manager and a watcher. We have this pattern in a few places. Can you potentially add comments in similar cases on what the manager is managing and what the watcher is watching?

Reply:
Sure, good idea.